x-pack/plugins/infra/public/pages/logs/settings/index_pattern_selector.tsx

```diff
@@ -28,15 +28,30 @@ export const IndexPatternSelector: React.FC<{
     fetchIndexPatternTitles();
   }, [fetchIndexPatternTitles]);
 
-  const availableOptions = useMemo<IndexPatternOption[]>(
-    () =>
-      availableIndexPatterns.map(({ id, title }) => ({
+  const availableOptions = useMemo<IndexPatternOption[]>(() => {
+    const options = [
+      ...availableIndexPatterns.map(({ id, title }) => ({
         key: id,
         label: title,
         value: id,
       })),
-    [availableIndexPatterns]
-  );
+      ...(indexPatternId == null || availableIndexPatterns.some(({ id }) => id === indexPatternId)
+        ? []
+        : [
+            {
```

Comment:
This synthesizes an entry in the selector to keep the selected state consistent even for broken references.